### PR TITLE
fix(factory-reset): umount filesystems after restore handling, before forced reboot

### PR DESCRIPTION
**Pull request by mlilien**

### CODE REVIEW

File: recipes-omnect/initrdscripts/omnect-os-initramfs/factory-reset [hunk 1]
```diff
@@ -112,6 +112,11 @@ factory_reset_finish() {
     # set return status
     set_bootloader_env_var ${FACTORY_RESET_STATUS_VAR} "${FACTORY_RESET_RET1}:${FACTORY_RESET_RET2}"
 
+    # if we had restore handling we have to clean unmount
+    if [ -n "${factory_reset_restore_list}" ]; then
```
Comment (HarryWaschkeit) [inline recipes-omnect/initrdscripts/omnect-os-initramfs/factory-reset:116]: does that really work?
I see variable factory_reset_restore_list being set and used in factory_reset_run() but don't see it defined as global variable in that file.

File: recipes-omnect/initrdscripts/omnect-os-initramfs/factory-reset [hunk 2]
```diff
@@ -112,6 +112,11 @@ factory_reset_finish() {
     # set return status
     set_bootloader_env_var ${FACTORY_RESET_STATUS_VAR} "${FACTORY_RESET_RET1}:${FACTORY_RESET_RET2}"
 
+    # if we had restore handling we have to clean unmount
```
[the remark this answers is not in the document]
Comment (mlilien) [reply recipes-omnect/initrdscripts/omnect-os-initramfs/factory-reset:115]: @JanZachmann

File: recipes-omnect/initrdscripts/omnect-os-initramfs/factory-reset (hunk 1 above)
Comment (mlilien) [reply recipes-omnect/initrdscripts/omnect-os-initramfs/factory-reset:116]: yes. it works reproducible. are not all non local variables, global? 
`factory_reset_finish()`  is called from within `factory_reset_run()` after setting `factory_reset_restore_list`. so it is in scope. do you want me to refactor it to use explicit local vars?

Comment (HarryWaschkeit) [reply recipes-omnect/initrdscripts/omnect-os-initramfs/factory-reset:116]: no, it's ok, until just now I never saw the implication that using a non-local variable in a function does set it always globally! I was only aware that an assignment in a function will change an already defined global variable.
but it makes perfect sense that this works either way, of course.

File: recipes-omnect/initrdscripts/omnect-os-initramfs/factory-reset (hunk 2 above)
[the remark this answers is not in the document]
Comment (JanZachmann) [reply recipes-omnect/initrdscripts/omnect-os-initramfs/factory-reset:115]: my 2ct:
In case of restore handling we have to clean unmount

**Review by JanZachmann — APPROVED** [review]

File: recipes-omnect/initrdscripts/omnect-os-initramfs/factory-reset (hunk 2 above)
[the remark this answers is not in the document]
Comment (HarryWaschkeit) [reply recipes-omnect/initrdscripts/omnect-os-initramfs/factory-reset:115]: never mind, I didn't read thoroughly (didn't see the "handling" for whatever reason :-(), it is ok as it is.

**Review by HarryWaschkeit — APPROVED** [review]
https://github.com/omnect/meta-omnect/pull/458/files#r1660562457

**Pull request merged**